e3sm_diags/driver/zonal_mean_2d_driver.py

```diff
@@ -216,7 +216,9 @@ def _get_avg_for_regridded_datasets(
     ds_diff_rg_avg = ds_diff_rg.spatial.average(var_key, axis=["X"])
 
     if parameter.diff_type == "relative":
-        ds_diff_rg_avg = ds_diff_rg_avg / ds_ref_rg_avg * 100.0
+        ds_diff_rg_avg[var_key] = (
+            ds_diff_rg_avg[var_key] / ds_ref_rg_avg[var_key] * 100.0
+        )
```

Comment:
Fixes relative diff plots displaying mean `np.nan`.

Example: https://web.lcrc.anl.gov/public/e3sm/cdat-migration-fy24/25-02-13-branch-930-zonal-mean-2d-mean-fix/viewer/zonal_mean_2d/era5-relative-difference/q-global-era5/ann.html

Reply:
This issue was caused by performing arithmetic on the entire regridded and spatially averaged diffs dataset, which also affected other variables such as bounds (resulting in 0 weighting and `np.nan` mean).